### PR TITLE
refactor(core): Decouple user events from internal hooks (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Follow-up to: #10280

### CODE REVIEW

File: packages/cli/src/controllers/me.controller.ts [hunk 1]
```diff
@@ -101,7 +99,6 @@ export class MeController {
 		this.authService.issueCookie(res, user, req.browserId);
 
 		const fieldsChanged = Object.keys(payload);
-		this.internalHooks.onUserUpdate({ user, fields_changed: fieldsChanged });
```
Comment (ivov) [inline packages/cli/src/controllers/me.controller.ts:104]: Pre-existing bug: Even when updating a single field, all fields exist in the payload, so all fields are being reported as having changed. To be fixed separately.

File: packages/cli/src/controllers/users.controller.ts [hunk 2]
```diff
@@ -294,11 +289,11 @@ export class UsersController {
 
 		await this.userService.update(targetUser.id, { role: payload.newRoleName });
 
-		this.internalHooks.onUserRoleChange({
-			user: req.user,
-			target_user_id: targetUser.id,
-			target_user_new_role: ['global', payload.newRoleName].join(' '),
```
Comment (ivov) [inline packages/cli/src/controllers/users.controller.ts:300]: Pre-existing bug: The target user's new role is being reported e.g. as 'global global:member'. To be fixed separately.

**Review by tomi — DISMISSED** [review]
Nice work 💯 Just questions about the follow up of those bugs discovered

File: packages/cli/src/controllers/me.controller.ts (hunk 1 above)
Comment (ivov) [reply packages/cli/src/controllers/me.controller.ts:104]: I'll follow up on this, won't forget

File: packages/cli/src/controllers/users.controller.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/controllers/users.controller.ts:300]: I'll follow up on this, won't forget

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.54.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.0)

**Pull request merged**